scripts/partition_manager_output.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+
+import argparse
+import yaml
+from os import path
+
+
+def get_header_guard_start(filename):
+    macro_name = filename.split('.h')[0]
+    return '''/* File generated by %s, do not modify */
+#ifndef %s_H__
+#define %s_H__''' % (__file__, macro_name.upper(), macro_name.upper())
+
+
+def get_header_guard_end(filename):
+    return "#endif /* %s_H__ */" % filename.split('.h')[0].upper()
+
+
+DEST_HEADER = 1
+DEST_KCONFIG = 2
+
+def get_config_lines(pm_config, head, split, dest):
+    config_lines = list()
+
+    def add_line(a, b):
+        return config_lines.append(head + "PM_" + a + split + b)
```

Comment:
Functions with side-effects should be avoided.

Could you have one function that just yields (key,value) tuples?

Reply:
My mistake, did not realize this was out-of-scope (copied code, not new code).

Reply:
I think it's fine to have side-effects when the function is defined within another function. The `return` seems redundant, though. Will remove.